server/lib/PLS/Server/Method/TextDocument.pm

```diff
@@ -126,6 +127,10 @@ sub get_request
     {
         return PLS::Server::Request::TextDocument::Completion->new($request);
     }
+    if ($method eq 'publishDiagnostics')
```

Comment:
Why did you add this? publishDiagnostics is a request from the server to the client - the client will never send this request, so it's not needed here.

Reply:
Ignore